framework/src/wifi_manager/wifi_manager.c

```diff
@@ -147,11 +163,14 @@ static void wifi_status_set(connect_status_e status)
 
 static void wifi_linkup_event_func(void)
 {
+	CHECK_WITHOUT_RETURN("linkup");
+
 	wifi_manager_cb_s *wifi_cb = g_manager_callback;
 	wifi_utils_info info;
 
 	wifi_mutex_acquire(w_info_mutex, WIFI_UTILS_FOREVER);
-	if (g_manager_info.mode == STA_MODE || (g_manager_info.mode == WIFI_MODE_CHANGING && g_manager_info.next_mode == STA_MODE)) {
+	if (g_manager_info.mode == STA_MODE || (g_manager_info.mode == WIFI_MODE_CHANGING && g_manager_info.next_mode == STA_MODE) || \
+		(g_manager_info.mode == WIFI_INITIALIZING && g_manager_info.next_mode == STA_MODE)) {
```

Comment:
Is the condition"(g_manager_info.mode == WIFI_INITIALIZING && g_manager_info.next_mode == STA_MODE)" necessary? When initializing, linkup handler is registered after calling WiFiStart function. So wifi_linkup_event_func function would not be called with WIFI_INITIALIZING mode.

Reply:
You're right. I remove this unnecessary condition.